src/ucp/dt/dt_iov.c

```diff
@@ -127,21 +122,16 @@ size_t ucp_dt_iov_count_nonempty(const ucp_dt_iov_t *iov, size_t iovcnt)
     return count;
 }
 
-static UCS_F_ALWAYS_INLINE ucs_status_t
-ucp_dt_iov_memtype_check(ucp_context_h context, const ucp_dt_iov_t *iov,
-                         size_t iovcnt, const ucp_request_param_t *param,
-                         ucp_memory_info_t *mem_info)
+ucs_status_t ucp_dt_iov_memtype_check(ucp_context_h context,
+                                      const ucp_dt_iov_t *iov, size_t iovcnt,
+                                      const ucp_memory_info_t *mem_info)
 {
     ucp_memory_info_t UCS_V_UNUSED mem_info_iter;
     int UCS_V_UNUSED i;
 
-    if (!ENABLE_PARAMS_CHECK) {
-        goto out;
-    }
-
     for (i = 0; i < iovcnt; ++i) {
-        ucp_memory_detect_param(context, iov[i].buffer, iov[i].length, param,
-                                &mem_info_iter);
+        ucp_memory_detect(context, iov[i].buffer, iov[i].length,
+                          &mem_info_iter);
```

Comment:
Maybe it is better to update the error message at line 137 too. `iov[0]` to smth like `expected`.

Reply:
imo better keep the message that indicates why we expect the memory type to be that value (because of iov[0])